provider/source/src/lib.rs

```diff
@@ -234,16 +234,21 @@ impl SourceDataProvider {
     ///
     /// ✨ *Enabled with the `networking` Cargo feature.*
     #[cfg(feature = "networking")]
-    pub fn with_icuexport_for_tag(self, mut tag: &str) -> Self {
-        if tag == "release-71-1" {
-            tag = "icu4x/2022-08-17/71.x";
-        }
+    pub fn with_icuexport_for_tag(self, tag: &str) -> Self {
+        let url = if tag >= "release-78.1" {
+            format!(
+                "https://github.com/unicode-org/icu/releases/download/{tag}/icu4x-exportdata-{}.zip",
+                tag.replace("release-", "")
+            )
+        } else {
+            format!(
+                "https://github.com/unicode-org/icu/releases/download/{tag}/icuexportdata_{}.zip",
+                tag.replace('/', "-")
+            )
+        };
```

Comment:
Suggestion:

- Keep the `/` to `-` substitution, in addition to the "release-" substitution
- Special-case "release-77.1" and any other releases that this datagen version is compatible with